### PR TITLE
Big rename of patterns to components in design system

### PR DESCRIPTION
**Pull request by andy-piccalilli**
Also featuring big ol' merge of patterns that got caught up in the problems on `main` yesterday

### CODE REVIEW

**Comment by andy-piccalilli** [comment]
@googlebot I consent

File: src/component-library/browser-compat/browser-compat.njk [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      <span class="tooltip__content">{{ item.label }}</span>
+      <div class="browser-compat__item" aria-hidden="true">
+        <img src="/images/icons/{{ item.image }}" alt="{{ item.name }}" />
+        <span class="indicator {{ item.indicator.utilities }}">{{ item.indicator.content }}</span>
+      </div>
+    </li>
+    {% endfor %}
+  </ul>
+	<p>
+		<a href="{{ data.link.url }}" target="_blank">{{ data.link.text }}</a>
```
Comment (devnook) [inline src/component-library/browser-compat/browser-compat.njk:15]: indentation

File: src/component-library/code-blocks/code-blocks.njk [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="flow">
+	<pre class="language-css"><code class="language-css"><span class="token selector">p</span> <span class="token punctuation">{</span><br>  <span class="token property">width</span><span class="token punctuation">:</span> 100px<span class="token punctuation">;</span><br>  <span class="token property">height</span><span class="token punctuation">:</span> 50px<span class="token punctuation">;</span><br>  <span class="token property">padding</span><span class="token punctuation">:</span> 20px<span class="token punctuation">;</span><br>  <span class="token property">border</span><span class="token punctuation">:</span> 1px solid<span class="token punctuation">;</span><br><span class="token punctuation">}</span></code></pre>
```
Comment (devnook) [inline src/component-library/code-blocks/code-blocks.njk:2]: indentation

File: src/site/content/en/design-system/welcome.njk [hunk 3]
```diff
@@ -26,7 +26,7 @@ This will create the following folder and file structure:
 ```
 web.dev
 └── src/
-    └── pattern-library/
+    └── components/
```
Comment (devnook) [inline src/site/content/en/design-system/welcome.njk:29]: shouldn't this be cmponent-library?

**Review by devnook — APPROVED** [review]
Only small nits left.

File: src/site/content/en/design-system/welcome.njk (hunk 3 above)
Comment (andy-piccalilli) [reply src/site/content/en/design-system/welcome.njk:29]: Component-library is the content url but makes sense to change the folder to that too tbh. Good catch

Comment (andy-piccalilli) [reply src/site/content/en/design-system/welcome.njk:29]: lol ignore me it already was 😂

**Comment by andy-piccalilli** [comment]
Lovely, ta!

**Comment by andy-piccalilli** [comment]
@googlebot I consent.

**Pull request merged**